ReactWindows/ReactNative/UIManager/DependencyObjectExtensions.cs

```diff
@@ -0,0 +1,73 @@
+﻿using ReactNative.Bridge;
+using System;
+using System.Runtime.CompilerServices;
+using Windows.UI.Xaml;
+
+namespace ReactNative.UIManager
+{
+    static class DependencyObjectExtensions
+    {
+        private static readonly ConditionalWeakTable<DependencyObject, DependencyObjectData> s_properties =
```

Comment:
We may want to consider making this a Dictionary and adding a cleanup mechanism.  It would give us more control over how / if a lock needs to be taken (since these operations are likely occurring on the dispatcher thread anyway).